scripts/firedrake-install

```diff
@@ -974,7 +974,14 @@ def build_and_install_petsc():
     else:
         log.info("Eigen tarball hash valid")
     log.info("Depending on your platform, PETSc may take an hour or more to build!")
-    assert not args.honour_petsc_dir  # Belt and braces security before we start clobbering things.
+    # Belt and braces security before we start clobbering things.
+    assert not args.honour_petsc_dir
+    if "HDF5_DIR" in os.environ:
+        log.info("""\nWarning: HDF5_DIR environment variable set, but
+ignored; to use your own HDF5, build your own PETSc
+with --with-hdf5-dir=$HDF5_DIR and then build firedrake
+with --honour-petsc-dir.
+""")
```

Comment:
I think I don't understand. Why would it not be OK for me to let Firedrake build PETSc and to set `PETSC_CONFIGURE_OPTIONS=--with-hdf5-dir=$HDF5_DIR`

Reply:
I thought `petsc_options` by default has `--download-hdf5`, so `--with-hdf5-dir=$HDF5_DIR` would cause conflict, but maybe we should just remove the former when we observe the latter?

Reply:
That would seem the logical thing to do.

Reply:
Updated and tested locally.